examples/decoding/cohy_decomposition.py

```diff
@@ -0,0 +1,557 @@
+
+print(
+    "\nTime to compute connectivity per epoch (window): ",
+    f"{duration/(N_EPOCHS//2):.0f} ms",
+)
+
+########################################################################################
+# Case 2: Fitting to and transforming the same data
+# -------------------------------------------------
+#
+# As mentioned above, the decoding module classes can also be used to transform the same
+# data the filters are fit to. This is a similar process to that of the
+# ``spectral_connectivity_...()`` functions, but with the increased efficiency of
+# fitting filters to a single frequency band as opposed to each frequency bin.
+#
+# To demonstrate this approach, we will load some example MEG data and divide it into
+# two-second-long epochs. We designate the left hemisphere sensors as the seeds and the
+# right hemisphere sensors as the targets. Since this is sensor-space data, we will use
+# the MIC method to analyse connectivity given its resilience to zero time-lag
+# interactions (See :doc:`../compare_coherency_methods` for more information).
+
+# %%
+
+raw = mne.io.read_raw_ctf(data_path() / "SubjectCMC.ds")
+raw.pick("mag")
+raw.crop(50.0, 110.0).load_data()
+raw.notch_filter(50)
+raw.resample(100)
+
+epochs = make_fixed_length_epochs(raw, duration=2.0).load_data()
+
+# left hemisphere sensors
+seeds = [idx for idx, ch_info in enumerate(epochs.info["chs"]) if ch_info["loc"][0] < 0]
+# right hemisphere sensors
+targets = [
+    idx for idx, ch_info in enumerate(epochs.info["chs"]) if ch_info["loc"][0] > 0
+]
+
+########################################################################################
+# There are two equivalent options for fitting and transforming the same data: 1)
+# passing the data to the :meth:`~mne_connectivity.decoding.CoherencyDecomposition.fit`
+# and :meth:`~mne_connectivity.decoding.CoherencyDecomposition.transform` methods
+# sequentially; or 2) using the combined
+# :meth:`~mne_connectivity.decoding.CoherencyDecomposition.fit_transform` method.
+#
+# We use the latter approach below, fitting the filters to the 15-20 Hz band and using
+# the ``"imcoh"`` method in the call to the ``spectral_connectivity_...()`` functions.
+# Plotting the results, we see a peak in connectivity at 15-20 Hz.
+
+# %%
+
+mic = CoherencyDecomposition(
+    info=epochs.info,
+    method="mic",
+    indices=(seeds, targets),
+    mode="multitaper",
+    fmin=FMIN,
+    fmax=FMAX,
+    rank=(3, 3),
+)
+
+start = time.time()
+epochs_transformed = mic.fit_transform(epochs.get_data())
+
+con_mic_class = spectral_connectivity_epochs(
+    epochs_transformed,
+    method="imcoh",
+    indices=mic.get_transformed_indices(),
+    fmin=5,
+    fmax=30,
+    sfreq=epochs.info["sfreq"],
+)
+class_duration = time.time() - start
+
+ax = plt.subplot(111)
+ax.plot(
+    con_mic_class.freqs,
+    np.abs(con_mic_class.get_data()[0]),
+    color=plt.rcParams["axes.prop_cycle"].by_key()["color"][2],
```

Comment:
```suggestion
    color="C2",
```
this is shorthand equivalent for "the third color in the cycle" (works for `"C0"` through `"C9"`)

Reply:
Can't believe I didn't know this...

Reply:
Can't believe I didn't know of this... Thanks!